### PR TITLE
[Fix]: Table tooltip issue

### PR DESCRIPTION
**Pull request by nxanil**
@killanch

Fixed issue #408. Please have a look.

### CODE REVIEW

File: Graphs/Table/index.js [hunk 1]
```diff
@@ -34,6 +37,7 @@ let container = '';
 let selectedRows = [];
 let displayColumn = [];
 let unformattedData = {};
+let currentStartIndex = 0;
```
Comment (killanch) [inline Graphs/Table/index.js:40]: lets avoid global, instead use state hook,
`const [currentStartIndex, setCurrentStartIndex] = useState(0);`

File: Graphs/Table/index.js [hunk 2]
```diff
@@ -667,7 +702,10 @@ const TableGraph = (props) => {
                                         return (
                                             <Table
                                                 ref={registerChild}
-                                                onRowsRendered={onRowsRendered}
+                                                onRowsRendered={({ startIndex }) => {
+                                                    currentStartIndex = startIndex;
+                                                    return onRowsRendered
```
Comment (killanch) [inline Graphs/Table/index.js:707]: this prop is a callback function.
to maintain parity with previous version, this should not return anything, instead should invoke InfiniteLoader's onRowsRendered()
```
onRowsRendered={(props) => {
    setCurrentStartIndex(props.startIndex);
    onRowsRendered(props);
}}
```

**Review by killanch — CHANGES_REQUESTED** [review]
@nxanil It is very difficult to navigate to the copy button, the tooltip disappears on mouse move, this needs to be addressed as well. Pls see sample below.

Behavior with code in this PR:
newtable

Old table behavior:
oldtable

**Comment by nxanil** [comment]
@killanch 

Feedback changes have been done. Please verify once.

File: Graphs/Table/index.js [hunk 3]
```diff
@@ -331,6 +337,41 @@ const TableGraph = (props) => {
...
+            getContent={[() => hoverContent(cellData, columnIndex, rowIndex)]}
+        />;
+    }
+
+    const cellRendererData = (cellData, columnIndex, rowIndex, column) => {
+        const data = (!!column.totalCharacters && !!cellData && cellData.length > column.totalCharacters) ? cellData.slice(0, column.totalCharacters) + '...' : cellData;
+
+        return (
+            <div>
+                <p data-tip='' data-for={`tooltip_${cellData}_${columnIndex}_${rowIndex}`} delayUpdate={1000}>{data}</p>
```
Comment (killanch) [inline Graphs/Table/index.js:369]: do we need this when tooltip is not applicable? also, is delayUpdate required within paragraph tag?
can we change this as,
```
return !!column.tooltip ?
    <React.Fragment>
        <p data-tip='' data-for={`tooltip_${cellData}_${columnIndex}_${rowIndex}`}>{data}</p>
        {renderTooltip(cellData, columnIndex, rowIndex)}
    </React.Fragment> : data;
```

**Review by killanch — COMMENTED** [review]
I also noticed that row gets selected when copy button is clicked, is there a way we can avoid row selection? this can be taken up in a separate PR
tabletooltip-copy

File: Graphs/Table/index.js [hunk 4]
```diff
@@ -331,6 +337,41 @@ const TableGraph = (props) => {
...
+                <button style={{ background: '#000', padding: 1 }} title="copy">
+                    <FaRegClipboard size={10} color="#fff" />
+                </button>
+            </CopyToClipboard>
+        </div>
+    }
+
+    const renderTooltip = (cellData, columnIndex, rowIndex) => {
+        const place = rowIndex === currentStartIndex ? 'bottom' : 'top';
+        const top = rowIndex === currentStartIndex ? (rowHeight * (rowIndex + 1)) - 20 : (rowHeight * rowIndex) - 15;
```
Comment (chaitanyahalbe) [inline Graphs/Table/index.js:353]: Can you please put a code comment on why you're deduction `20` and `15`? Those numbers seem arbitrary without any comment.

**Comment by nxanil** [comment]
@killanch 

Feedback changes have been done. Please verify once.

**Review by killanch — COMMENTED** [review]
@nxanil looks good, thanks

**Review by killanch — APPROVED** [review]

**Pull request merged**